### ISSUE TITLE
Importing sql instance with no provider project fails

### ISSUE THREAD

**Issue by hawksight** [terraform-providers/terraform-provider-google#1410]
### Terraform Version
```
Terraform v0.11.7
+ provider.google v1.9.0
+ provider.null v1.0.0
+ provider.random v1.2.0
+ provider.template v1.0.0
```

### Affected Resource(s)
- google_sql_database_instance
- Google provider

### Terraform Configuration Files
```hcl
provider "google" {
  # Terraform svc account creds
  credentials = "${file(var.GCLOUD_KEYFILE_JSON)}"
  region      = "us-central1"
  version     = "~> 1.9"
}

...

module "sql" {
  source      = "../terraform-modules/gcloud-sql-single"
  project     = "${module.project.PROJECT_ID}"
  name_prefix = "${module.project.PROJECT_ID}"
  region      = "${var.G_REGION}"
  zone        = "us-central1-a"
  env         = "dev"
}
```

Module just has `random_int` and `google_sql_database_instance` resources:
```hcl
resource "random_integer" "rand" {
  max = 9999
  min = 0001
}

resource "google_sql_database_instance" "single" {
  # Resource required
  region           = "${var.region}"
  # Module
  project          = "${var.project}"
  database_version = "${lookup(var.conf[var.env] , "database_version")}"
  name             = "${var.name_prefix}-${random_integer.rand.result}"

  settings {
    # Resource required
    tier              = "${lookup(var.conf[var.env] , "tier")}"
    # Module
    activation_policy = "${lookup(var.conf[var.env] , "activation_policy")}"
    availability_type = "ZONAL"
    disk_autoresize   = "${lookup(var.conf[var.env] , "disk_autoresize")}"
    disk_size         = "${lookup(var.conf[var.env] , "disk_size")}"

    ip_configuration {
      require_ssl  = "true"
    }

    location_preference {
      zone         = "${var.zone}"
    }

    maintenance_window {
      day          = "${lookup(var.conf[var.env] , "maint_day")}"
      hour         = "${lookup(var.conf[var.env] , "maint_hour")}"
      update_track = "stable"
    }

    backup_configuration {
      enabled      = "${lookup(var.conf[var.env] , "backup_enabled")}"
      start_time   = "${lookup(var.conf[var.env] , "backup_start_time")}"
    }
  }
}
```

### Debug Output
N/A

### Panic Output
N/A

### Expected Behavior
Terraform import should have imported the resource as normal with the command:
`terraform import -var-file=inputs.tfvars module.sql.google_sql_database_instance.single sopost-infra-dev-4552`

### Actual Behaviour
Terraform produced an error stating the `project` was not set:
```
module.sql.google_sql_database_instance.single: Importing from ID "sopost-infra-dev-4552"...
module.sql.google_sql_database_instance.single: Import complete!
  Imported google_sql_database_instance (ID: sopost-infra-dev-4552)
module.sql.google_sql_database_instance.single: Refreshing state... (ID: sopost-infra-dev-4552)

Error: module.sql.google_sql_database_instance.single (import id: sopost-infra-dev-4552): 1 error(s) occurred:

* import module.sql.google_sql_database_instance.single result: sopost-infra-dev-4552: google_sql_database_instance.single: project: required field is not set
```

### Steps to Reproduce

1. Have an existing database instance.
2. Create a terraform config with provider (do not provide a default `project`) and the resource `google_sql_database_instance`.
3. Try to import the sql instance to the resource definition you created.

(I assume this would reproduce, but have not tried exactly this way)

### Important Factoids

In the configuration I am creating the project and everything within. Therefore I have opted not to provide the project in the `provider` configuration:
1. Because terraform wouldn't let me interpolate it back from the `google_project` resource
2. It makes no sense to have a default when you are creating the project, and can use the ID from that.

I am using a service account from one project, to create and setup an entirely new project.

Context: I only had to import because google's api produced an error with a permissions error for the SQL api 'accessNotConfigured'. This was just after enabling. Upon trying to create again I was surprised to see it was created already, so the name could not be reused... hence attempting to import.

#### Workaround
By going back and setting the `project` in the provider configuration and attempting the import again did work.

### References
- Seems similar to this #973 

I think that during the import terraform should only look to the `provider` for configuration when the configuration is not, or cannot be set in the `resource` configuration. As you can see, the `project` field has been set on the `google_sql_database_instance` resource and should have been passed in by the module call.

**Comment by paddycarver** [comment]
All of your diagnosis seems correct, to me. I just want to comment on this part:

> I think that during the import terraform should only look to the provider for configuration when the configuration is not, or cannot be set in the resource configuration. As you can see, the project field has been set on the google_sql_database_instance resource and should have been passed in by the module call.

When importing a resource, at the moment, the provider only gets access to the ID. That is, if you're running `terraform import module.sql.google_sql_database_instance.single sopost-infra-dev-4552` the provider only gets `sopost-infra-dev-4552` and the type of resource it is, and has to identify it based on that. So even though the config has the information we need, at the moment, we still need to change that import to look more like `terraform import -var-file=inputs.tfvars module.sql.google_sql_database_instance.single my-project/sopost-infra-dev-4552` so we have access to the project, which lets us find the database instance.

So, we can fix this, but unfortunately that will just mean adding extra info when importing the resource, not pulling info from the config.

**Comment by hawksight** [comment]
I had not realised the import resource doesn't actually use the resources configuration, although now you say it, something does ring a bell.

Your proposal sounds good:
```
terraform import -var-file=inputs.tfvars <terraform_resource> <project>/<resource_name>
```
But I assume that this might break how import works with other providers? 
Can't think is AWS uses projects or equivalent. But if this doable in just google space it would be great.

The `<project>/` bit could be optional too, so as not to break current usage for lots of people.
I have the pragmatic solution for now, so its not urgent to fix.

**Comment by hawksight** [comment]
just a though whilst I was browsing elsewhere, how about passing the `self_link` in rather than the name? self links usually contain all that valuable googley information like project and region etc. and should be easily parsable?

**Comment by paddycarver** [comment]
This should have been fixed by #1956. I'm going to close it out. If someone is still seeing this problem, feel free to comment, and we can reopen and investigate.
